spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -246,12 +247,23 @@ private DeleteOrphanFiles.Result doExecute() {
     List<String> orphanFiles =
         findOrphanFiles(spark(), actualFileIdentDS, validFileIdentDS, prefixMismatchMode);
 
-    Tasks.foreach(orphanFiles)
-        .noRetry()
-        .executeWith(deleteExecutorService)
-        .suppressFailureWhenFinished()
-        .onFailure((file, exc) -> LOG.warn("Failed to delete file: {}", file, exc))
-        .run(deleteFunc::accept);
+    if (deleteFunc != null || !(table.io() instanceof SupportsBulkOperations)) {
```

Comment:
nit: What about inverting this condition instead of using negation?

```
if (deleteFunc == null && table.io() instanceof SupportsBulkOperations) {
  // bulk deletes
} else {
  // non-bulk delete
}
```